daemon/status.go

```diff
@@ -32,9 +34,305 @@ import (
 )
 
 const (
+	// collectStatusInterval is how often statuses will be collected
 	collectStatusInterval = 5 * time.Second
+
+	// staleTimeout is how stale a status can be before we report an error
+	staleTimeout = 1 * collectStatusInterval
```

Comment:
Maybe `2 * collectStatusInterval`?

Reply:
Discussed offline. best option would be one timeout per collector. I'm fine with 30s or 1 minute